src/operators.jl

```diff
@@ -136,3 +147,7 @@ function inducedsubgraph{T<:AbstractGraph}(g::T, iter)
     end
     return h, newvid
 end
+
+# dispatch for g[[1,2,3]], g[1:3], g[Set([1,2,3])]
+# these are the only allowed dispatches, everything else is slow
+getindex(g::AbstractGraph, iter) = first(induced_subgraph(g, iter))
```

Comment:
When this is called you are throwing away the mapping of old names to new names. Is that because in the intended use case, people using g[vs] are only operating on the subgraph? I guess people who want the mapping can just use the induced_subgraph interface.